test/plugin/base_test.rb

```diff
@@ -142,6 +142,24 @@ def test_metadata_agent_url_customization
     end
   end
 
+  def test_configure_ignores_unknown_monitoring_type
+    # Verify that driver creation succeeds when monitoring type is not
+    # "prometheus" (in which case, we simply don't record metrics),
+    # and that the counters are set to nil.
+    setup_gce_metadata_stubs
+    create_driver(CONFIG_UNKNOWN_MONITORING_TYPE)
+    assert_nil(Prometheus::Client.registry.get(
```

Comment:
I'm inclined to suggest factoring this out into an `assert_no_prometheus_metric` method, to abstract away the registry lookup, but I'd like @qingling128 to sanity-check this request.
If you do factor it out, this test would read as follows:
```ruby
    setup_gce_metadata_stubs
￼    create_driver(CONFIG_UNKNOWN_MONITORING_TYPE)
    assert_no_prometheus_metric(:stackdriver_successful_requests_count)
    assert_no_prometheus_metric(:stackdriver_failed_requests_count)
    assert_no_prometheus_metric(:stackdriver_ingested_entries_count)
    assert_no_prometheus_metric(:stackdriver_dropped_entries_count)
    assert_no_prometheus_metric(:stackdriver_retried_entries_count)
```

Reply:
Either way works for me. I like how clean it looks this way. Yet it's not a lot of duplicated code.

Reply:
I like to keep all direct accesses to dependent libraries in one place, rather than sprinkled through the tests, but won't insist if you are ok with leaving it as-is.